torchvision/prototype/models/alexnet.py

```diff
@@ -29,12 +29,14 @@ class AlexNetWeights(Weights):
 
 
 def alexnet(weights: Optional[AlexNetWeights] = None, progress: bool = True, **kwargs: Any) -> AlexNet:
+    if type(weights) == bool and weights:
+        _deprecated_positional(kwargs, "pretrained", "weights", True)
     if "pretrained" in kwargs:
-        warnings.warn("The parameter pretrained is deprecated, please use weights instead.")
-        weights = AlexNetWeights.ImageNet1K_RefV1 if kwargs.pop("pretrained") else None
+        weights = _deprecated_param(kwargs, "pretrained", "weights", AlexNetWeights.ImageNet1K_RefV1)
```

Comment:
We could add an additional parameter to the decorator named `deprecated_param_idx` and also check `*args` if it is passed.

On a side note, this is one of the reasons I aggressively prefer keyword-only arguments for every parameter that doesn't have an obvious meaning. `pretrained` is not such a case since I would have no idea what `alexnet(True)` does without reading documentation. 

I understand we want to keep BC here, but maybe we can have a deprecation period and change to keyword argument only afterwards? I can write a decorator for that if you like.

Reply:
Yes agreed, everyone is in agreement here that we should prefer keyword arguments and we should deprecate the old usage. We need to coordinate this to avoid pushing too many changes concurrently but I'm fully supportive of doing this.